docker/release/config/opensearch/opensearch-docker-entrypoint.sh

```diff
@@ -49,21 +45,23 @@ function setupSecurityPlugin {
         else
             echo "Enabling OpenSearch Security Plugin"
         fi
+    else
+        echo "OpenSearch Security Plugin does not exist, disable by default"
     fi
 }
 
-# Trap function that is used to terminate opensearch and performance analyzer
-# when a relevant signal is caught.
-function terminateProcesses {
-    if kill -0 $OPENSEARCH_PID >& /dev/null; then
-        echo "Killing opensearch process $OPENSEARCH_PID"
-        kill -TERM $OPENSEARCH_PID
-        wait $OPENSEARCH_PID
-    fi
-    if kill -0 $PA_PID >& /dev/null; then
-        echo "Killing performance analyzer process $PA_PID"
-        kill -TERM $PA_PID
-        wait $PA_PID
+# Performance Analyzer Plugin
+function setupPerformanceAnalyzerPlugin {
+    PERFORMANCE_ANALYZER_PLUGIN="opensearch-performance-analyzer"
+    if [ -d "$OPENSEARCH_HOME/plugins/$PERFORMANCE_ANALYZER_PLUGIN" ]; then
+        if [ "$DISABLE_PERFORMANCE_ANALYZER_AGENT_CLI" = "true" ]; then
+            echo "Disabling execution of $OPENSEARCH_HOME/bin/$PERFORMANCE_ANALYZER_PLUGIN/performance-analyzer-agent-cli for OpenSearch Performance Analyzer Plugin"
+        else
+            echo "Enabling execution of OPENSEARCH_HOME/bin/$PERFORMANCE_ANALYZER_PLUGIN/performance-analyzer-agent-cli for OpenSearch Performance Analyzer Plugin"
+            $OPENSEARCH_HOME/bin/opensearch-performance-analyzer/performance-analyzer-agent-cli > $OPENSEARCH_HOME/logs/performance-analyzer.log 2>&1 &
+        fi
+    else
+        echo "OpenSearch Performance Analyzer Plugin does not exist, disable by default"
     fi
```

Comment:
Shouldn't we retain this logic to terminate in case signal is received?

Reply:
The whole point of this PR is, if PA terminated do not even change the behavior of OS.
OS can keep running while PA terminated.

The initial PA/OS coupled logic is out of date now, it was meant to replicate supervisord behavior previously.